### ISSUE TITLE
XDocReportRegistry cache all reports by default and can cause memory leak

### ISSUE THREAD

**Issue by peyo** [opensagres/xdocreport#229]
Hello,

We use XDocReport with an application and it works fine, but we have a misfortune because all reports are keept in memory (500 times the same template). 

Two factors at this problem : 
 - when we use XDocReportRegistry.getRegistry().loadReport(InputStream sourceStream, String reportId) cache option used is "true"
 - the javadoc not warn us about this behavior

I suggest (it's not a bug) to deactivate caching option by default. I think it's not a logical behavior
` public IXDocReport loadReport( InputStream sourceStream, String reportId )
        throws IOException, XDocReportException
    {
        return loadReport( sourceStream, reportId, false );
    }`

Or create explicit function, example : "loadReportToCache".

Bye

**Comment by angelozerr** [comment]
I agree with you, please do a PR.

**Comment by peyo** [comment]
Of course, now we do this but it's after we have an outOfMemory in production.

**Comment by peyo** [comment]
Excuse my precedent message the bad fonction is not : 
`public IXDocReport loadReport( InputStream sourceStream, String reportId )`
But 
`public IXDocReport loadReport( InputStream sourceStream, ITemplateEngine templateEngine )`

When we used it, template are cached.

**Comment by peyo** [comment]
If you want I make a PR for fix that.

**Comment by nguyenhuybo** [comment]
in version 2.0.1 

IXDocReport report = XDocReportRegistry.getRegistry().loadReport(in, TemplateEngineKind.Velocity);  caused Memory leak

Because loadReport() default cached template
return loadReport( sourceStream, templateEngineKind, true );  <--- cache

in version 2.0.2
IXDocReport report = XDocReportRegistry.getRegistry().loadReport(in, TemplateEngineKind.Velocity); 
 default cache is false
return loadReport( sourceStream, templateEngineKind, false ); <--- not cache anymore

But to use cache then do the following
IXDocReport report = XDocReportRegistry.getRegistry().loadReport(in, TemplateEngineKind.Velocity, true);

This time the template is still cached but not caused MEMORY LEAK anymore 

Thank you :-D
